### ISSUE TITLE
Set up continuous integration

### ISSUE THREAD

**Issue by cmacmackin** [ukaea/Indica#2]
Tests should be run for all pushes to the repository, especially on merge requests. This could be done using GitHub actions or another CI service.

The tricky part of this is how to do tests that use actual JET data. The European Commission doesn't like the data being exposed publicly, so we can't store it in the repository. This is particularly problematic for tests of the `PPFReader` class and likely for integration tests as well.

Possible solutions:
- Generate fake data to use instead (tedious and error-prone)
- Connect the CI server to the UKAEA VPN so it can access JET data (clumsy, possibly a security risk?)
- Have a test runner that lives at UKAEA (don't know if that is even possible)

**Comment by cmacmackin** [comment]
I've set up the workflows for running tests with GitHub actions. The problem of how to test the PPF reader remains, however. There was some suggestion that suitably mangled (or maybe encrypted) data might be allowed, but this would require sign-off from the lawyers.

**Comment by cmacmackin** [comment]
See [this explaination of encrypting data for GitHub Actions](https://docs.github.com/en/actions/reference/encrypted-secrets#limits-for-secrets).

**Comment by cmacmackin** [comment]
I have configured the tests so that they will run without time limits on CI. I needed to do this as the CI runs more slowly than my development machine and tests which were within the deadline locally would fail for exceeding the deadline on CI.

I've configured the tests only to run those with a chance of passing (i.e., which are testing a feature that has been implemented and which are not themselves broken in some way, such as in #11). This means that we should actually be able to catch regressions now that they aren't being drowned out by tests which always fail! As part of this I have also configured the PPF reader tests only to run if the fake PPF data is available. Hopefully we can still make this available in the CI somehow, however.
